### PR TITLE
Add random command to fetch a random snap

### PR DESCRIPTION
**Pull request by bumblepie**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -130,6 +130,15 @@ bot.on('message', function (message) {
         return;
       }
       switch (lCommand) {
+        case 'random':
+          let results = scraps.find({ 'botMessageId': { '$nin': [null, 'IN_PROGRESS'] } });
+          let count = results.length;
+          let id = Math.floor(Math.random() * count);
+          let scrap = results[id];
+          message.channel.send(`Here is a random snap XD`).then(msg => {
```
Comment (SarenCurrie) [inline index.js:138]: XD

File: index.js [hunk 2]
```diff
@@ -130,6 +130,15 @@ bot.on('message', function (message) {
         return;
       }
       switch (lCommand) {
+        case 'random':
+          let results = scraps.find({ 'botMessageId': { '$nin': [null, 'IN_PROGRESS'] } });
```
Comment (SarenCurrie) [inline index.js:134]: What does this bit do? `null, 'IN_PROGRESS'`

Comment (SarenCurrie) [reply index.js:134]: I see. It's some state that's there already. /shrug

**Review by SarenCurrie — APPROVED** [review]
lgtm

File: index.js (hunk 2 above)
Comment (bumblepie) [reply index.js:134]: From what I can see, it filters out snaps that are in the progress of being saved to the db and so don't have an id yet. See line 496-506 ish

**Pull request merged**